packages/components/src/internal/actions.ts

```diff
@@ -1813,126 +1844,59 @@ function applySelection(
-    if (!col.isPublicLookup()) {
-        return false;
+    if (lookupKeyValues) {
+        selectRowsOptions.filterArray = [Filter.create(column.lookup.keyColumn, lookupKeyValues, Filter.Types.IN)];
     }
 
-    const lookup = getLookupStore(col);
-
-    if (!lookup) {
-        return true;
-    } else if (!lookup.isLoading && !lookup.isLoaded) {
-        return true;
-    } else if (values && !lookup.containsAll(values)) {
-        return true;
-    }
-
-    return false;
-}
-
-export function searchLookup(
-    column: QueryColumn,
-    maxRows: number,
-    token?: string,
-    values?: List<string>,
-    keys?: List<any>
-) {
-    let store = getLookupStore(column);
-
-    // prevent redundant search
-    if (store && (token !== store.lastToken || values || keys)) {
-        store = updateLookupStore(store, {
-            isLoaded: false,
-            isLoading: true,
-            lastToken: token,
-            loadCount: store.loadCount + 1,
-        });
-
-        const lookup = column.lookup;
-
-        const selectRowOptions: any = {
-            schemaName: lookup.schemaName,
-            queryName: lookup.queryName,
-            columns: [lookup.displayColumn, lookup.keyColumn].join(','),
-            containerPath: lookup.containerPath,
-            maxRows,
-            includeTotalCount: 'f',
-        };
-
-        if (values) {
-            selectRowOptions.filterArray = [
-                Filter.create(column.lookup.displayColumn, values.toArray(), Filter.Types.IN),
-            ];
-        }
-
-        if (keys) {
-            selectRowOptions.filterArray = [Filter.create(column.lookup.keyColumn, keys.toArray(), Filter.Types.IN)];
-        }
-
-        return searchRows(selectRowOptions, token, lookup.displayColumn)
-            .then(result => {
-                const { displayColumn, keyColumn } = column.lookup;
-                const { key, models, totalRows } = result;
-
-                if (models[key]) {
-                    const descriptors = fromJS(models[key])
-                        .reduce((list, row) => {
-                            const key = row.getIn([keyColumn, 'value']);
-
-                            if (key !== undefined && key !== null) {
-                                return list.push({
-                                    display:
-                                        row.getIn([displayColumn, 'displayValue']) ||
-                                        row.getIn([displayColumn, 'value']),
-                                    raw: key,
-                                });
-                            }
-                        }, List<ValueDescriptor>())
-                        .sortBy(vd => vd.display, naturalSort)
-                        .reduce((map, vd) => map.set(vd.raw, vd), OrderedMap<any, ValueDescriptor>());
+    const result = await selectRows(selectRowsOptions);
+    const {displayColumn, keyColumn} = column.lookup;
```

Comment:
minor, but you could move this up to the top of the function and then just use `displayColumn` and `keyColumn` in more places instead of `column.lookup. displayColumn`